.devcontainer/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM mcr.microsoft.com/devcontainers/cpp:1-ubuntu-22.04
```

Comment:
@wusatosi , would you like to add a new doc in `https://github.com/beman-project/beman/blob/main/docs/DEV.md` where we can add a small section explaining what is codespaces and how can you use it? A short video can be put there (or GIF).

We can then put a link in this repo, in `docs/dev.md`, reference to the one from beman repo. Check something similar in https://github.com/beman-project/optional26/blob/main/docs/debug-ci.md. (I also think this doc should be move into beman/)

Reply:
yes, sure, I will work on this later in the week.

There's a lot of catch-up exemplar need to do on the documentation side, in fact that's my this-week's-plan!

I will file documentation on another PR, leaving this for functionality in case someone come in and ask for a major change.